be/src/testing/testing.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, MethodNotAllowedException } from '@nestjs/common';
+import { ItemService } from '../item/item.service';
+import { CreateItemInput } from 'src/item/dto/create-item.input';
+import { CategoryService } from 'src/category/category.service';
+import { CreateCategoryInput } from 'src/category/dto/create-category.input';
+import { CreateNoteInput } from 'src/note/dto/create-note.input';
+import { NoteService } from 'src/note/note.service';
+import { UserService } from 'src/user/service/user.service';
+import { CreateUserDTO } from 'src/user/dto/user.create.dto';
+import { Role } from 'src/user/roles/user.roles';
+import * as faker from 'faker';
+
+@Injectable()
+export class TestingService {
+  constructor(
+    private readonly categoryRepo: CategoryService,
+    private readonly itemRepository: ItemService,
+    private readonly noteRepo: NoteService,
+    private readonly userRepo: UserService,
+  ) {
+    if (process.env.ADD_FAKE_DATA === 'TRUE') this.AddFakeData();
+    console.log('add fake data ', process.env.ADD_FAKE_DATA);
+  }
+
+  AddFakeData() {
+    if (process.env.ADD_FAKE_DATA !== 'TRUE') {
+      throw new MethodNotAllowedException(`ADD_FAKE_DATA not set to TRUE`);
+    }
+    for (let i = 0; i < 3; i++) {
+      const item: CreateItemInput = { regCode: faker.datatype.number() };
+      this.itemRepository.createItem(item);
+
+      const category: CreateCategoryInput = {
+        name: faker.name.findName(),
+        bundelable: faker.datatype.boolean(),
+        parentCategoryId: null,
+      };
+      this.categoryRepo.create(category);
+
+      const note: CreateNoteInput = { content: faker.lorem.text(), itemId: 1 };
+      this.noteRepo.createNote(note);
+
+      const user: CreateUserDTO = {
+        username: faker.internet.userName(),
+        password: faker.internet.password(),
+        email: faker.internet.email(),
+        phone: faker.phone.phoneNumber(),
+        full_name: faker.name.findName(),
+        role: Math.floor(Math.random() * 2) === 1 ? Role.Admin : Role.User,
```

Comment:
The likelihood of this `=== 1` of being true is extremely small

Reply:
No, Math.random() * 2 generates a number between 0 and 2 then Math.floor method rounds it down to an integer. This generates either 0 or 1.

Reply:
ah, yep. I didn't notice the `Math.floor`, my bad